Modules/Microsoft365DSC/DSCResources/MSFT_TeamsGroupPolicyAssignment/MSFT_TeamsGroupPolicyAssignment.psm1

```diff
@@ -0,0 +1,433 @@
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftTeams' -InboundParameters $PSBoundParameters
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    #check policyname
+    $command = "get-cs" + $PolicyType
+    $Policys = Invoke-Expression -Command $command -ErrorAction SilentlyContinue
+    $policymatch = $false
+    if($null -ne $Policys){
+        Foreach($policy in $Policys.Identity){
+            $match = "^Tag:" + $PolicyName + "$"
+            if($policy -match $match){
+                $policymatch = $true
+            }
+        }
+    }
+    if($null -eq $Policys -or $policymatch -eq $false){
+        Write-Verbose -Message "No PolicyType found for $PolicyType"
+        return
+    }
+
+    #get groupid
+    if($GroupId.Length -eq 0)
+    {
+        $Group = Find-CsGroup -SearchQuery $GroupDisplayname
+        if($group.Length -gt 1)
+        {
+            Write-Verbose -Message "Found $($group.Length) groups with the name $GroupDisplayname"
+            $Group = $Group | Where-Object { $_.DisplayName -eq $GroupDisplayname }
+        }
+        if($null -eq $Group)
+        {
+            Write-Verbose -Message "Group not found for $GroupDisplayname"
+            return
+        }
+        $GroupId = $Group.Id
+    }
+    Write-Verbose -Message "Retrieve GroupId for: $($GroupDisplayname)"
+    try{
+        if($Ensure -eq 'Present' -and $CurrentValues.Ensure -eq 'Absent')
+        {
+            Write-Verbose -Message "Adding GroupPolicyAssignment for $GroupDisplayname"
+            new-csGroupPolicyAssignment -GroupId $GroupId -PolicyType $PolicyType -PolicyName $PolicyName -Rank $Priority
+        }
+        elseif($Ensure -eq 'Present' -and $CurrentValues.Ensure -eq 'Present')
+        {
+            #Set-CsGroupPolicyAssignment not implemented jet / use remove-add as described in docs
+            Write-Verbose -Message "Remove GroupPolicyAssignment for $GroupDisplayname"
+            Remove-CsGroupPolicyAssignment -GroupId $CurrentValues.GroupId -PolicyType $CurrentValues.PolicyType
+            Write-Verbose -Message "Adding GroupPolicyAssignment for $GroupDisplayname"
+            new-csGroupPolicyAssignment -GroupId $GroupId -PolicyType $PolicyType -PolicyName $PolicyName -Rank $Priority
+        }
+        elseif($Ensure -eq 'Absent' -and $CurrentValues.Ensure -eq 'Present')
+        {
+            Write-Verbose -Message "Remove GroupPolicyAssignment for $GroupDisplayname"
+            Remove-CsGroupPolicyAssignment -GroupId $CurrentValues.GroupId -PolicyType $CurrentValues.PolicyType
+        }
+    }
+    catch{
+        Write-Verbose -Message "Error while setting GroupPolicyAssignment for $GroupDisplayname"
```

Comment:
Please use the same logging logic as used on line 106 to 110. That will also output to the event log and other logs.